### DISCUSSION TITLE
If I move to "localhost:3000/de", I can't go back go default language. It forces reroute to "/de"

### DISCUSSION THREAD

**Discussion by matt18465** [amannn/next-intl#804] [Q&A]
### Is your feature request related to a problem? Please describe.

I have language dropdown with NextLink. If I'm on "localhost:3000/de", when I click "/" link it should move to "/" but it doesn't. It just stays on "/de" all the time.  I have 2 languages: "en" and "de". English is default, I also have option enabled to hide default locale from domain. 

**The same is if I don't use my dropdown and just type "localhost:3000" in new tab. It reroutes to "localhost:3000/de"**

When I open "localhost:3000" from anonomous mode in browser, it doesn't reroute to "/de", as expected

reproduction:

1. Open "localhost:3000"
2. Move to "localhost:3000/de"
3. Try to go back to "localhost:3000"

### Describe the solution you'd like

When I open "localhost:3000/de", then move to "localhost:3000" (either by NextLink or manually writing adress in new tab), it should open "localhost:3000" as expected, not "localhost:3000/de".

```
`import { notFound } from "next/navigation";
import { getRequestConfig } from "next-intl/server";

// Can be imported from a shared config
const locales = ["en", "de"];

export default getRequestConfig(async ({ locale }) => {
  // Validate that the incoming `locale` parameter is valid
  if (!locales.includes(locale as any)) notFound();

  return {
    messages: (await import(`../messages/${locale}.json`)).default,
  };
});
`

`import createMiddleware from "next-intl/middleware";

export default createMiddleware({
  // A list of all locales that are supported
  locales: ["en", "de"],

  // Used when no locale matches
  defaultLocale: "en",
  localePrefix: 'as-needed'
});

export const config = {
  // Match only internationalized pathnames
  matcher: ["/", "/(en|pl)/:path*"],
};
```
`

### Describe alternatives you've considered

no alternatives

**Comment by aXenDeveloper** [comment]
> I have language dropdown with NextLink.

You can't use `Link` from `Next`. See more => https://next-intl-docs.vercel.app/docs/routing/navigation#link

**Comment by amannn** [comment]
You might want to check out [the middleware docs](https://next-intl-docs.vercel.app/docs/routing/middleware) in detail to learn about how to change languages and which redirects are applied.

I'll move this to a discussion since it's a usage question.

**Reply by janoschherrmann** [reply]
Hey, similar issue here (using `localePrefix: 'as-needed'`), but the middleware docs didn't seem to solve my question. In my case, I want to navigate to `/` by editing the URL in my browser, but this redirect to `/[currentLocale]` still happens, instead of changing the local to the default locale which should be available through `/`.

If made sure that the middleware matches unprefixed paths as well, but that didn't help: 
`export const config = {
  matcher: '/((?!api|_next|_vercel|.*\\..*).*)',
};`
